gdsfactory/read/from_yaml.py

```diff
@@ -1059,6 +1060,7 @@ def _update_reference_by_placement(
     port = p.port
     rotation = p.rotation
     mirror = p.mirror
+    mirror_x = getattr(p, "mirror_x", None)
```

Comment:
I think this is a really bad idea to just allow new yaml keys; it should rather just be a function in the pydantic classes. This mirror can be expressed as a modified placement with mirror already